events/event.py

```diff
@@ -2,21 +2,35 @@
 
 import logging
 
+from events.utils import import_object
+
 
 class Event(object):
     """Class to reperesent an event."""
 
     def __init__(self, name):
         self.name = name
         self._listeners = []
+        self._methods = None
 
     def add_listener(self, l, action):
         self._listeners.append((l, action))
 
+    @property
+    def methods(self):
+        if self._methods is None:
+            self._methods = []
+            for listener, action in self._listeners:
+                mod_name, obj_name = listener.rsplit('.', 1)
+                klass = import_object(mod_name, obj_name)
+                method = klass(action=action)
+                self._methods.append((action, method))
+        return self._methods
+
     def emit(self, *args, **kwargs):
         logger = logging.getLogger('events')
-        for l, action in self._listeners:
+        for name, method in self.methods:
             try:
-                l(action=action)(*args, **kwargs)
+                method(*args, **kwargs)
```

Comment:
We should still instantiate the class every time the event is emitted.

Reply:
True, but this is sufficient to avoid circular dependencies when importing top level modules.